tests/tests.rs

```diff
@@ -229,50 +229,49 @@ async fn test_on_disk_index() -> Result<()> {
     common::clean(storage, path).await
 }
 
-#[tokio::test]
-async fn test_work_dir_lock() {
-    use nix::sys::wait::{waitpid, WaitStatus};
-    use nix::unistd::{fork, ForkResult};
-
-    let path = common::init("work_dir_lock");
+#[test]
+fn test_work_dir_lock() {
+    use rusty_fork::{fork, rusty_fork_id};
+    use std::sync::Arc;
+    use tokio::runtime::Builder;
 
+    let path = Arc::new(common::init("work_dir_lock"));
+    let child_path = path.clone();
     // We need separate processes because locks do not work within the same process
-    match unsafe { fork() } {
-        Ok(ForkResult::Parent { child }) => {
+    fork(
+        "test_work_dir_lock",
```

Comment:
It would be good to not hardcode the function name. I found this crate: https://docs.rs/function_name/latest/function_name/ . It adds a macro to get the name of the function

Reply:
According to the documentation, you should wrap the function name with `rusty_fork_test_name!` macro (https://docs.rs/rusty-fork/latest/rusty_fork/macro.rusty_fork_test_name.html)